src/implements.jl

```diff
@@ -16,7 +16,10 @@ all the mandatory components of the interace are implemented.
 """
 function implements end
 implements(T::Type{<:Interface}, obj) = implements(T, typeof(obj))
-implements(::Type{<:Interface}, obj::Type) = false
+implements(T::Type{<:Interface}, obj::Type) = inherits(T, obj)
+
+function inherits end
```

Comment:
I'm not qualified to review the exact implem but there are lots of codecov misses here